### PR TITLE
Auto path with catalog 

### PR DESCRIPTION
**Pull request by profxj**
As titled.  

This PR introduces a script to auto-run PATH on an input localization.

Currently supports the 'Pan-STARRS' and 'DECaL' surveys.

Would like to add unit tests..

### CODE REVIEW

File: astropath/scripts/use_catalogs.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+    # Save?
+    if pargs.outfile is not None:
+        Path.candidates.to_csv(pargs.outfile, index=False)
+        print(f"Wrote: {pargs.outfile}")
+
+    # Return
+    return Path
+
+# Test
```
Comment (SunilSimha) [inline astropath/scripts/use_catalogs.py:146]: Can this documentation be made to show with the parser help text?

File: astropath/scripts/use_catalogs.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    elif pargs.survey == 'DECaL':
+        mag_key = 'DECaL_r'
+        # Cuts
+        cut_mag = (catalog[mag_key] > 14.) & np.isfinite(catalog[mag_key]) 
+        cut_star = catalog['gaia_pointsource'] == 0
+        keep = cut_mag & cut_star
+        # Half-light radius
+        catalog['ang_size'] = np.maximum(catalog['shapedev_r'], catalog['shapeexp_r'])
+        zero = catalog['ang_size'] == 0.
+        catalog['ang_size'][zero] = 1. # KLUDGE!!
```
Comment (SunilSimha) [inline astropath/scripts/use_catalogs.py:86]: Probably should warn users if one of these galaxies gets a high PATH probability. I'd add a flag to the catalog that keeps track of this and warns later if P_Ox is high for one of these sources.

File: astropath/scripts/use_catalogs.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+    from astropy import units
+
+    from frb.surveys import survey_utils
+
+    from astropath import path
+    from astropath.scripts.utils import coord_arg_to_coord
+    from astropath.utils import radec_to_coord
+
+    scale = 0.5
```
Comment (SunilSimha) [inline astropath/scripts/use_catalogs.py:44]: Should this be user adjustable?

File: astropath/scripts/use_catalogs.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    Path.init_candidates(catalog['ra'],
+                         catalog['dec'],
+                         catalog['ang_size'],
+                         mag=catalog[mag_key])
+
+    # Candidate prior
+    Path.init_cand_prior('inverse', P_U=pargs.PU)
+
+    # Offset prior
+    Path.init_theta_prior('exp', 6., scale)
```
Comment (SunilSimha) [inline astropath/scripts/use_catalogs.py:124]: Should `theta_max` also be user supplied or at least an optional argument?

**Review by SunilSimha — CHANGES_REQUESTED** [review]
Looks good for the most part. I'm requesting one warning to be implemented and I've left a few comments. Will write unit tests for these shortly.

File: astropath/scripts/use_catalogs.py (hunk 1 above)
Comment (profxj) [reply astropath/scripts/use_catalogs.py:148]: instead, I generated this:

https://github.com/FRBs/astropath/blob/auto_path_with_catalog-/docs/scripts.rst

File: astropath/scripts/use_catalogs.py (hunk 2 above)
Comment (profxj) [reply astropath/scripts/use_catalogs.py:88]: good suggestion. I tried doing something like this but it was painful.

Am tripping a warning instead..

File: astropath/scripts/use_catalogs.py (hunk 3 above)
Comment (profxj) [reply astropath/scripts/use_catalogs.py:44]: yes, good call.  done

**Review by SunilSimha — APPROVED** [review]
Looks good!

**Pull request merged**